python/plugins/processing/algs/qgis/RasterSampling.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    RasterSampling.py
+    -----------------------
+    Date                 : July 2018
+    Copyright            : (C) 2018 by Matteo Ghetta
+    Email                : matteo dot ghetta at gmail dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Matteo Ghetta'
+__date__ = 'July 2018'
+__copyright__ = '(C) 2018, Matteo Ghetta'
+
+# This will get replaced with a git SHA1 when you do a git archive
+
+__revision__ = '$Format:%H$'
+
+
+import os
+
+from qgis.PyQt.QtGui import QIcon
+from qgis.PyQt.QtCore import QVariant
+
+from qgis.core import (QgsApplication,
+                       QgsField,
+                       QgsFeatureSink,
+                       QgsRaster,
+                       QgsProcessing,
+                       QgsProcessingParameterRasterLayer,
+                       QgsProcessingParameterString,
+                       QgsFields,
+                       QgsProcessingUtils,
+                       QgsProcessingException,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterFeatureSink)
+
+from processing.algs.qgis.QgisAlgorithm import QgisAlgorithm
+
+
+class RasterSampling(QgisAlgorithm):
+
+    INPUT = 'INPUT'
+    RASTERCOPY = 'RASTERCOPY'
+    COLUMN_PREFIX = 'COLUMN_PREFIX'
+    OUTPUT = 'OUTPUT'
+
+    def name(self):
+        return 'rastersampling'
+
+    def displayName(self):
+        return self.tr('Sample Raster Values')
+
+    def group(self):
+        return self.tr('Raster analysis')
+
+    def groupId(self):
+        return 'rasteranalysis'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(
+            QgsProcessingParameterFeatureSource(
+                self.INPUT,
+                self.tr('Input Point Layer'),
+                [QgsProcessing.TypeVectorPoint]
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterRasterLayer(
+                self.RASTERCOPY,
+                self.tr('Raster Layer to sample'),
+            )
+        )
+
+        self.addParameter(
```

Comment:
I'd suggest making this an advanced parameter. (see e.g. https://github.com/qgis/QGIS/blob/c1552e728a27f8cd4c807d63a301a125d6934188/python/plugins/processing/algs/qgis/Heatmap.py#L111 )